### PR TITLE
Misc cleanup

### PR DESCRIPTION
**Pull request by tamird**
Minor cleanup I stumbled upon.

### CODE REVIEW

File: kv/dist_sender.go [hunk 1]
```diff
@@ -402,7 +402,7 @@ func (ds *DistSender) sendRPC(desc *proto.RangeDescriptor,
 	}
 
 	// Copy and rearrange the replicas suitably, then return the desired order.
-	replicas := proto.ReplicaSlice(append([]proto.Replica(nil), desc.Replicas...))
+	replicas := append(proto.ReplicaSlice{}, desc.Replicas...)
```
Comment (petermattis) [inline kv/dist_sender.go]: Note that `proto.ReplicaSlice{}` and `proto.ReplicaSlice(nil)` are different. The former does an allocation creating a non-nil but empty slice. The latter does not.

File: kv/dist_sender.go [hunk 2]
```diff
@@ -423,16 +423,16 @@ func (ds *DistSender) sendRPC(desc *proto.RangeDescriptor,
...
 		if err != nil {
 			if log.V(1) {
-				log.Infof("node %d address is not gossiped: %v", replicas[i].NodeID, err)
+				log.Infof("node %d address is not gossiped: %v", replica.NodeID, err)
 			}
 			continue
 		}
 		addrs = append(addrs, addr)
-		replicaMap[addr.String()] = &replicas[i]
+		replicaMap[addr.String()] = &replica
```
Comment (petermattis) [inline kv/dist_sender.go]: I think this is incorrect. `&replica` will have the same value throughout the loop. I wonder why no tests are failing.

File: kv/dist_sender.go (hunk 1 above)
Comment (tamird) [inline kv/dist_sender.go]: Understood. Does `(append(proto.ReplicaSlice(nil), desc.Replicas...)` differ from `append(proto.ReplicaSlice{}, desc.Replicas...)` in the total number of allocations?

Comment (petermattis) [inline kv/dist_sender.go]: I don't know. In general, I use `proto.ReplicaSlice(nil)` when I want an empty slice.

**Comment by tamird** [comment]
Alright, down to just the one commit

**Comment by bdarnell** [comment]
LGTM, but what about the decision to not make minor stylistic changes so that `git blame` is preserved? :)

**Comment by tamird** [comment]
I know you're trolling, but I think this is a simplification, not just a stylistic change :)

**Pull request merged**